pymc/sampling.py

```diff
@@ -1687,6 +1708,20 @@ def compile_forward_sampling_function(
     if basic_rvs is None:
         basic_rvs = []
 
+    if constant_data is None:
+        constant_data = {}
+    if constant_coords is None:
+        constant_coords = set()
+
+    # We define a helper function to check if shared values match to an array
+    def shared_value_matches(var):
+        shared_value = var.get_value()
+        try:
+            array = constant_data[var.name]
+            return np.array_equal(shared_value, array)
+        except KeyError:
+            return var.name in constant_coords
```

Comment:
Reordering for performance, and also setting `borrow=False`, otherwise it deepcopies the internal value unnecessarily 

```suggestion
        try:
            old_array_value = constant_data[var.name]
        except KeyError:
            return var.name in constant_coords
        current_shared_value = var.get_value(borrow=True)
        return np.array_equal(old_array_value, current_shared_value)
```

Reply:
Much nicer!

Reply:
Btw your last push does not seem to have included any changes

Reply:
Oh stupid me!